ffi/capi/src/units_converter.rs

```diff
@@ -69,13 +70,30 @@ pub mod ffi {
     pub struct ICU4XMeasureUnitParser<'a>(pub MeasureUnitParser<'a>);
 
     impl<'a> ICU4XMeasureUnitParser<'a> {
+        /// Parses the CLDR unit identifier (e.g. `meter-per-square-second`) and returns the corresponding [`ICU4XMeasureUnit`].
+        /// Returns an error if the unit identifier is not valid or the `unit_id` is not a valid UTF-8 string.
+        #[diplomat::rust_link(
+            icu::experimental::units::measureunit::MeasureUnitParser::parse,
+            FnInStruct
+        )]
+        pub fn parse_unit_identifier<'text>(
+            &self,
+            unit_id: &'text DiplomatStr,
+        ) -> Result<Box<ICU4XMeasureUnit>, ICU4XError> {
+            let unit_id = core::str::from_utf8(unit_id)
+                .map_err(|_| ICU4XError::InvalidCldrUnitIdentifierError)?;
+            Ok(Box::new(ICU4XMeasureUnit(
+                self.0.try_from_identifier(unit_id)?,
+            )))
+        }
+
         /// Parses the CLDR unit identifier (e.g. `meter-per-square-second`) and returns the corresponding [`ICU4XMeasureUnit`].
         /// Returns an error if the unit identifier is not valid.
         #[diplomat::rust_link(
             icu::experimental::units::measureunit::MeasureUnitParser::parse,
             FnInStruct
         )]
-        pub fn parse_measure_unit(
+        pub fn parse_valid_utf8_unit_identifier(
```

Comment:
I think our convention here is to use
```suggestion
        pub fn parse_unit_identifier_valid_utf8(
```